### PR TITLE
feat: Create Menu component

### PR DESCRIPTION
**Pull request by sebpearce**
## Changes

- Creates a new draft component called `Menu`. (We had discussions about the name – `dropdown`, `dropdown-menu` are already names of draft packages, which limited the choices here.)
- This component mixes together the existing `Dropdown` and `MenuList` components into one thing, while exposing the Menu stuff for consumers to build their own menus.
- It allows consumers to pass in any Button they want, including IconButtons (such as kebabs/meatball icons).

image

## Background

A number of engineers have never been happy with the dropdown/menu situation. We currently have two components, `Dropdown` and `MenuList`. `MenuList` is just the content that fills the dropdown menu container, and `Dropdown` provided the container and the clickable button/control action/icon.

A major problem with `Dropdown` is that it duplicated its own visual clickable action, instead of letting the user pass in whatever they wanted. This meant that even after we uplifted the `Button` component, dropdown buttons still looked inconsistent and out of date.

Also, the separation of `Dropdown` and `MenuList` (along with an Elm version in a separate package called `dropdown-menu`) is potentially confusing for consumers.

## UI Kit design

The designs are based on the Menu sticker sheet in the UI Kit, [available here](https://www.figma.com/file/GMxm8rvDCbj0Xw3TQWBZ8b/Zen-UI-Kit?node-id=6262%3A0):

image

### CODE REVIEW

File: draft-packages/menu/KaizenDraft/Menu/Menu.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { Button } from "@kaizen/draft-button"
+import { cleanup, fireEvent, render } from "@testing-library/react"
+import * as React from "react"
+import Menu from "./Menu"
+
+afterEach(cleanup)
+
+describe("Dropdown", () => {
+  test("renders default view", () => {
```
Comment (ActuallyACat) [inline draft-packages/menu/KaizenDraft/Menu/Menu.spec.tsx:9]: nit/ In other tests, we tend to use `it` rather than `test`

Comment (sebpearce) [reply draft-packages/menu/KaizenDraft/Menu/Menu.spec.tsx:9]: Happy to change this after merging if that's OK with you. These tests were copied from the existing Dropdown and I just removed a few that were no longer applicable.

Comment (sebpearce) [reply draft-packages/menu/KaizenDraft/Menu/Menu.spec.tsx:9]: (The merge gives me access to the MenuItemProps which unblocks other work)

**Comment by sebpearce** [comment]
Sorry I should have mentioned btw, a lot of these were copied directly from MenuList + Dropdown, which is why there is a bit of weirdness (BEM structure in CSS etc.). I am happy to chip away at all that as I work on it, but that's why it's there.

File: draft-packages/menu/package.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
+{
+  "name": "@kaizen/draft-menu",
+  "version": "1.5.5",
```
Comment (ActuallyACat) [inline draft-packages/menu/package.json:3]:
```suggestion
  "version": "1.0.0",
```

Comment (sebpearce) [reply draft-packages/menu/package.json:3]: 🤦‍♂️

File: draft-packages/menu/KaizenDraft/Menu/components/MenuItem.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    <span className={styles.menuItem__Icon}>
+      <Icon icon={icon} role="presentation" />
+    </span>
+  )
+
+  const className = classNames({
+    [styles.menuItem]: true,
+    [styles["menuItem--destructive"]]: destructive,
+    [styles["menuItem--disabled"]]: disabled,
+  })
```
Comment (ActuallyACat) [inline draft-packages/menu/KaizenDraft/Menu/components/MenuItem.tsx:30]:
```suggestion
  const className = classNames({
    styles.menuItem,
    styles["menuItem--destructive"]: destructive,
    styles["menuItem--disabled"]: disabled,
  })
```

Comment (ActuallyACat) [reply draft-packages/menu/KaizenDraft/Menu/components/MenuItem.tsx:30]: Oops, that's on the wrong line ^ Y'know what I mean though.

File: draft-packages/menu/KaizenDraft/Menu/Menu.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
+import * as React from "react"
+// TODO: get this from @kaizen/draft-button once published version offers ButtonProps
+import { ButtonProps } from "../../../button"
```
Comment (ActuallyACat) [inline draft-packages/menu/KaizenDraft/Menu/Menu.tsx:3]: I can't approve this until this import has been changed. When this is packaged up this button will throw an error, as it doesn't exist.

File: draft-packages/menu/KaizenDraft/Menu/components/MenuItem.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+  const className = classNames({
+    [styles.menuItem]: true,
+    [styles["menuItem--destructive"]]: destructive,
+    [styles["menuItem--disabled"]]: disabled,
+  const className = classNames({
+    styles.menuItem,
+    styles["menuItem--destructive"]: destructive,
+    styles["menuItem--disabled"]: disabled,
+  })
```
Comment (sebpearce) [inline draft-packages/menu/KaizenDraft/Menu/components/MenuItem.tsx:34]:
```suggestion
  const className = classNames({
    styles.menuItem,
    styles["menuItem--destructive"]: destructive,
    styles["menuItem--disabled"]: disabled,
  })
```

**Pull request merged**